pkg/ipam/poolallocator/allocator.go

```diff
@@ -266,3 +289,98 @@ func (a *IPPoolAllocator) Release(ip net.IP) error {
 	}
 	return err
 }
+
+// ReleaseResource releases the IP associated with specified Pod. It returns error if the resource is not present in state or in case CRD failed to update its state.
+// In case of success, IP pool CRD status is updated with released entry.
+func (a *IPPoolAllocator) ReleasePod(namespace, podName string) error {
```

Comment:
We expect those to be used by garbage collector

Reply:
I think GC might not need those methods as it should look for orphaned Pods so `HasPod` won't be useful and it's likely that it will remove orphaned Pods in one call to be efficient instead of one by one.
But I'm fine with changing it later when GC is implemented.